backend/mem/file.go

```diff
@@ -0,0 +1,469 @@
+package mem
+
+import (
+	"errors"
+	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v4/utils"
+	"io"
+	"path"
+	"time"
+)
+
+//File implements vfs.File interface for os fs.
+type File struct {
+	exists    bool
+	timeStamp time.Time
+	isOpen    bool
+	isRef     bool         //has it been referenced before?
+	privSlice []byte       //the file contents
+	Filename  string       //the base name of the file
+	cursor    int          //the index that the buffer (privSlice) is at
+	location  vfs.Location //the location that the file exists on
+}
+
+/*		******* Error Functions *******		*/
+func doesNotExist() error {
+	return errors.New("This file does not exist!")
+}
+func copyFail() error {
+	return errors.New("This file was not successfully copied")
+}
+
+func copyFailNil() error {
+	return errors.New("The target file passed in was nil")
+}
+
+func deleteError() error {
+	return errors.New("Deletion was unsuccessful")
+}
+
+func moveToLocationError() error {
+	return errors.New("Move to location unexpectedly failed")
+}
+func writeError() error {
+	return errors.New("Unexpected Write Error")
+}
+
+func seekError() error {
```

Comment:
The way these are used is actually hiding the underlying error. You are catching the error and returning this error in your functions without wrapping the original error. It's probably better just to return the error directly in most cases.